src/module/iotjs_module_dns.c

```diff
@@ -15,30 +15,36 @@
 
 #include "iotjs_def.h"
 
+#include "iotjs_module_dns.h"
+
 #include "iotjs_reqwrap.h"
 #include "uv.h"
 
 
-typedef struct {
-  iotjs_reqwrap_t reqwrap;
-  uv_getaddrinfo_t req;
-} IOTJS_VALIDATED_STRUCT(iotjs_getaddrinforeqwrap_t);
-
-
 #define THIS iotjs_getaddrinforeqwrap_t* getaddrinforeqwrap
 
-void iotjs_getaddrinforeqwrap_initialize(THIS, const iotjs_jval_t* jcallback) {
+iotjs_getaddrinforeqwrap_t* iotjs_getaddrinforeqwrap_create(
+    const iotjs_jval_t* jcallback) {
+  iotjs_getaddrinforeqwrap_t* getaddrinforeqwrap =
+      IOTJS_ALLOC(iotjs_getaddrinforeqwrap_t);
   IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_getaddrinforeqwrap_t,
                                      getaddrinforeqwrap);
-  iotjs_reqwrap_initialize(&_this->reqwrap, jcallback, (uv_req_t*)&_this->req,
-                           getaddrinforeqwrap);
+  iotjs_reqwrap_initialize(&_this->reqwrap, jcallback, (uv_req_t*)&_this->req);
+  return getaddrinforeqwrap;
 }
 
 
-void iotjs_getaddrinforeqwrap_destroy(THIS) {
+static void iotjs_getaddrinforeqwrap_destroy(THIS) {
```

Comment:
Thanks for your review, @glistening. I agree that it will be better to declare the function without `THIS` macro. (It was added just to shorten the line.)

However, since this PR is not directly related to `THIS` macro, if you're OK with the rest of this PR, I think it will be better to close this one, and create a new one to remove `THIS` macros. Would it be OK with you?